xm-commons-timeline/src/main/java/com/icthh/xm/commons/timeline/config/DomainEventTimelineInterceptor.java

```diff
@@ -0,0 +1,141 @@
+package com.icthh.xm.commons.timeline.config;
+
+import com.icthh.xm.commons.domainevent.domain.DomainEvent;
+import com.icthh.xm.commons.domainevent.domain.HttpDomainEventPayload;
+import com.icthh.xm.commons.domainevent.domain.enums.DefaultDomainEventSource;
+import com.icthh.xm.commons.domainevent.service.EventPublisher;
+import com.icthh.xm.commons.logging.util.MdcUtils;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.http.HttpHeaders;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.apache.commons.lang3.ObjectUtils.firstNonNull;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class DomainEventTimelineInterceptor extends HandlerInterceptorAdapter {
+
+    private static final String HEADER_TENANT = "x-tenant";
+    private static final String AUTH_TENANT_KEY = "tenant";
+    private static final String AUTH_USER_KEY = "user_key";
+
+    private final EventPublisher eventPublisher;
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) throws Exception {
+
+        if (isIgnoredRequest(request, response)) {
+            return;
+        }
+
+        final OAuth2Authentication auth = getAuthentication();
+        if (auth == null) {
+            String tenant = request.getHeader(HEADER_TENANT);
+            publishEvent(request, response, tenant, null, null);
+        } else {
+            Map<String, String> details = getUserDetails(auth);
+            String tenant = details.getOrDefault(AUTH_TENANT_KEY, "");
+            String userKey = details.getOrDefault(AUTH_USER_KEY, "");
+            String userLogin = (String) auth.getPrincipal();
+            publishEvent(request, response, tenant, userLogin, userKey);
+        }
+    }
+
+    private OAuth2Authentication getAuthentication() {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        if (auth instanceof OAuth2Authentication) {
```

Comment:
pls don't cast to OAuth2Authentication, this class removed in master branch, add all needed method to xm auth context